pix2pix-ultimate/metrics.py

```diff
@@ -1,44 +1,270 @@
-import os
-from glob import glob
+import numpy as np
+import scipy.misc
+from scipy.ndimage import _ni_support
+from scipy.ndimage.morphology import distance_transform_edt, binary_erosion,\
+    generate_binary_structure
+
 from utils import *
 
 
-# Return all images of one slice of a brain (seperated)
-def get_img_row(overall_img, img_size, i):
-    from_ = i * img_size
+predefined_threshold = 1.0
+
+
+def true_positive(gt_image, predicted_image, threshold):
+    """ Returns the amount of pixel which are positive in both images.
+
+    :param gt_image: ground truth image
+    :param predicted_image: predicted image
+    :param threshold: threshold to make binary decision
+    :return: amount of pixel which are positive in both images
+    """
+    return np.count_nonzero(gt_image[predicted_image >= threshold])
+
+
+def true_negative(gt_image, predicted_image, threshold):
+    """ Returns the amount of pixel which are negative in both images.
+
+    :param gt_image: ground truth image
+    :param predicted_image: predicted image
+    :param threshold: threshold to make binary decision
+    :return: amount of pixel which are negative in both images
+    """
+    return np.count_nonzero((gt_image < threshold)[(predicted_image < threshold)])
+
+
+def false_positive(predicted_image, true_positive, threshold):
+    """ Returns the amount of pixel which are negative in the ground truth but positive in predicted image.
+
+    :param predicted_image: predicted image
+    :param true_positive: amount of correct positive pixels
+    :param threshold: threshold to make binary decision
+    :return: amount of pixel which are negative in the ground truth but positive in prediction image
+    """
+    return abs(np.count_nonzero(predicted_image >= threshold) - true_positive)
+
+
+def false_negative(gt_image, threshold, true_positive):
```

Comment:
Could you move the threshold to the last position of the function parameters? That would make it more consistent with the other function signatures